pkg/modules/auth/openid/openid.go

```diff
@@ -280,13 +293,44 @@ func getOrCreateUser(s *xorm.Session, cl *claims, provider *Provider, idToken *o
 		if cl.Name != u.Name {
 			u.Name = cl.Name
 		}
+		if cl.Picture != "" {
+			u.AvatarProvider = "openid"
+		}
 		u, err = user.UpdateUser(s, u, false)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	return
+	if cl.Picture != "" && (isNewUser || u.AvatarProvider == "openid" || u.AvatarProvider == "") {
+		log.Debugf("Found avatar URL for user %s: %s", u.Username, cl.Picture)
+
+		// Download avatar
+		avatarData, err := utils.DownloadImage(cl.Picture)
+		if err != nil {
+			log.Errorf("Error downloading avatar: %v", err)
+			// Continue, do not interrupt the authentication process due to avatar download failure
+		} else {
```

Comment:
Can you return early instead of the `else`?

Reply:
return u, nil will immediately end the function, skipping any future code. Current design allows adding more code after avatar sync and will not affect avatar sync failure

Reply:
I mean, that's a problem when we're adding code after this in the future. After thinking about it, can you refactor this part out into a separate function? Then it should be no problem to return early.